### PR TITLE
#857 Pressing 'Upload' without attachment chosen redirects to home page 

### PR DESCRIPTION
**Pull request by kujtimiihoxha**
For #857
 - Disabled upload button if there is no file selected.

### CODE REVIEW

**Comment by karato** [comment]
@kujtimiihoxha I will find a reviewer for your pull requests shortly, thanks for contribution!

**Comment by karato** [comment]
@mkordas do this review

**Comment by mkordas** [comment]
@kujtimiihoxha I'm on it

**Comment by kujtimiihoxha** [comment]
@mkordas Here is a screenshot of how it looks now if you don't have any file selected.
screen1

**Comment by kujtimiihoxha** [comment]
@mkordas ping.

**Comment by mkordas** [comment]
@kujtimiihoxha Sorry for delay, thanks for pinging

File: netbout-web/src/main/js/bout.js [hunk 1]
```diff
@@ -251,6 +251,19 @@ $(document).ready(
           return false;
         }
     );
+  if ($( "#file-binary" )[0].value === "") {
+      $( "#file-submit" ).prop('disabled', true);
+  } else {
+      $( "#file-submit" ).prop('disabled', false);
+  }
+  $( "#file-binary" ).change(function() {
```
Comment (mkordas) [inline netbout-web/src/main/js/bout.js]: @kujtimiihoxha let's apply paired brackets rule here too, `function() {` should be on new line

File: netbout-web/src/main/js/bout.js [hunk 2]
```diff
@@ -251,6 +251,19 @@ $(document).ready(
...
+      if($( "#file-binary" )[0].value === ""){
```
Comment (mkordas) [inline netbout-web/src/main/js/bout.js]: @kujtimiihoxha do we need these spaces inside `$(  )`?

File: netbout-web/src/main/js/bout.js [hunk 3]
```diff
@@ -251,6 +251,19 @@ $(document).ready(
           return false;
         }
     );
+  if ($( "#file-binary" )[0].value === "") {
```
Comment (mkordas) [inline netbout-web/src/main/js/bout.js]: @kujtimiihoxha don't we need any tests for that?

File: netbout-web/src/main/js/bout.js [hunk 4]
```diff
@@ -251,6 +251,19 @@ $(document).ready(
...
+      $( "#file-submit" ).prop('disabled', true);
+  } else {
+      $( "#file-submit" ).prop('disabled', false);
+  }
+  $( "#file-binary" ).change(function() {
+      if($( "#file-binary" )[0].value === ""){
+          $( "#file-submit" ).prop('disabled', true);
+      } else {
```
Comment (mkordas) [inline netbout-web/src/main/js/bout.js]: @kujtimiihoxha can we somehow avod duplication of the if statement?

**Comment by mkordas** [comment]
@kujtimiihoxha see my comments above

File: netbout-web/src/main/js/bout.js (hunk 3 above)
Comment (kujtimiihoxha) [inline netbout-web/src/main/js/bout.js]: @mkordas I do not see any tests for javascript in netbout.

**Comment by kujtimiihoxha** [comment]
@mkordas removed duplicate and change the function.

**Comment by mkordas** [comment]
@kujtimiihoxha OK, checking again...

File: netbout-web/src/main/js/bout.js [hunk 5]
```diff
@@ -251,6 +257,8 @@ $(document).ready(
           return false;
         }
     );
+  $("#file-binary").change(checkUpload);
+  checkUpload();
```
Comment (mkordas) [inline netbout-web/src/main/js/bout.js]: @kujtimiihoxha does it have proper indentation?

File: netbout-web/src/main/js/bout.js [hunk 6]
```diff
@@ -177,7 +177,13 @@ function scrollOrLoad(amountRecords) {
     }
   }
 }
-
```
Comment (mkordas) [inline netbout-web/src/main/js/bout.js]: @kujtimiihoxha I see that other functions are separated by newline here

File: netbout-web/src/main/js/bout.js [hunk 7]
```diff
@@ -177,7 +177,13 @@ function scrollOrLoad(amountRecords) {
...
+function checkUpload() {
```
Comment (mkordas) [inline netbout-web/src/main/js/bout.js]: @kujtimiihoxha can you think of more descriptive name? E.g. `updateUploadButtonState` or `checkIfUploadShouldBeDisabled`.

File: netbout-web/src/main/scss/bout.scss [hunk 8]
```diff
@@ -175,3 +175,7 @@
     }
   }
 }
+input[type="submit"]:disabled {
```
Comment (mkordas) [inline netbout-web/src/main/scss/bout.scss]: @kujtimiihoxha other stuff is separated by newline here

**Comment by mkordas** [comment]
@kujtimiihoxha please see 4 minor comments more

File: netbout-web/src/main/js/bout.js (hunk 3 above)
Comment (mkordas) [inline netbout-web/src/main/js/bout.js]: @kujtimiihoxha OK, thanks

**Comment by kujtimiihoxha** [comment]
@mkordas  Done.

**Comment by mkordas** [comment]
@kujtimiihoxha nice

**Comment by mkordas** [comment]
@rultor merge

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by karato** [comment]
@elenavolokhova please, review this task for compliance with our [quality rules](http://at.teamed.io/qa.html)

**Comment by karato** [comment]
@rultor deploy now pls

**Comment by elenavolokhova** [comment]
@karato Looks good!

**Comment by karato** [comment]
> @karato Looks good!

@elenavolokhova thank you

**Comment by karato** [comment]
@mkordas quality is good here, 10 mins added to @dmzaytsev (our architect) in transaction 78637250

**Comment by karato** [comment]
@mkordas added 10 mins to @elenavolokhova, for QA review, payment code is `78642681`. Much appreciated! **31 mins** was added to your account, payment ID `56d1ed11313c2b790a0004e0`, time spent is 46 hours and 25 mins. you're getting extra minutes here (c=16). added +31 to your rating, now it is equal to [+7459](http://www.netbout.com/b/36080?open=rating)

**Pull request merged**